### DISCUSSION TITLE
Add async state support for Combobox

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#1208] [Ideas] [answered]
Hello team!

We've been using combobox more and more to replace our `react-select` components and one thing I wasn't able to migrate was loading state. The loading state in `react-select` provides users with feedback see that their input changes are being processed.

https://react-select.com/async

![image](https://user-images.githubusercontent.com/6936736/108559585-58873a80-72b0-11eb-9fbc-0f676407aeb9.png)

**Comment by SiTaggart** [comment, marked as answer]
Hi @vnguyen94 

You can just use a single custom option in `items` for this, whist you're wait for a response. Once you've got a response update the items array with the new data.

**Reply by vankhoawin** [reply]
I feel like this ties in with my previous question about table loading states https://github.com/twilio-labs/paste/discussions/862. We could technically build this into each async Combobox we use sure, but isn't this a pattern considered widespread enough through Twilio to be built-in to the component itself? We still have custom loaders for each async table (and i'm sure other teams are either doing the same thing or don't have loading states altogether)

**Reply by SiTaggart** [reply]
We won't likely bake in the functionality into the component. 

What we will likely do is:

1) When we get to rewriting the combobox docs we'll include a tonne of examples and patterns on how to use it for various situations. There's simply too many teams and too many variables to create a one size fits all, but we can provide the guidelines, patterns and examples.

2) When we have skeleton loaders, we'll add loading examples and patterns to the various components that need them. Again, we likely won't build in loading states to the components, just give guidance on the different ways to do it.

**Reply by vankhoawin** [reply]
sounds good, thanks for following up
